File: cpp/GFG_DSA/1.Maths/IsPalindrome.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

bool IsPalindrome(int n)
{
    int rev = 0, original = n;
    while (n != 0)
    {
        rev = rev * 10 + n % 10;
        n /= 10;
    }
    return original == rev;
}

int main()
{
    //region READ FROM FILE
    freopen("../input.txt", "r", stdin);
    freopen("../output.txt", "w", stdout);
    //end region

    int n;
    cin >> n;
    cout << IsPalindrome(n);

    return 0;
}

File: python/gfg/3.Recurrsion/isPalindrome.py
def IsPalindrome(s, start, end):
    if start >= end:
        return True
    return s[start] == s[end] and IsPalindrome(s, start + 1, end - 1)


s = input()
print(IsPalindrome(s, 0, len(s) - 1))
